### PR TITLE
Some exception hygeine in RoomsDAO, Unit tests.

### PR DESCRIPTION
**Pull request by kb000**
Added Junit 4 unit tests which run in maven and eclipse.
This merge looks like someone (maybe me?) doesn't have autocrlf set to true, so every file seems changed.

### CODE REVIEW

File: src/com/comeet/UserService.java [hunk 1]
```diff
@@ -1,251 +1,250 @@
...
+    @Path("/{orgDomain}/rooms/{roomRecipient}/reserve")
+    @Produces("application/json")
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+    public List<Meeting> reserveRoom(@Context HttpHeaders headers,
+                    @PathParam("orgDomain") String orgDomain,
+                    @PathParam("roomRecipient") String roomRecipient,
+                    @FormParam("subject") String subject, @FormParam("body") String body,
+                    @FormParam("start") String start, @FormParam("end") String end,
+                    @DefaultValue("") @FormParam("required") String requiredRecipients,
+                    @DefaultValue("") @FormParam("optional") String optionalRecipients)
```
Comment (Dairai) [inline src/com/comeet/UserService.java:150]: we don't handle optional attendees?

Comment (kb000) [reply src/com/comeet/UserService.java:150]: I mean, we drop them on the floor.  This adheres to the apiary contract.

**Review by Dairai — APPROVED** [review]
Looks good to me. we definitely should apply unit tests to our individual code. I will get mine started

**Comment by kb000** [comment]
[INFO] Building war: /var/lib/jenkins/workspace/RESTApi Pull Request Validation/target/comeet.war
Java HotSpot(TM) 64-Bit Server VM warning: INFO: os::commit_memory(0x00000000fa941000, 47071232, 0) failed; error='Cannot allocate memory' (errno=12)
#
# There is insufficient memory for the Java Runtime Environment to continue.

**Comment by kb000** [comment]
buddybuild now

**Pull request merged**